qiskit/synthesis/evolution/product_formula.py

```diff
@@ -151,7 +151,7 @@ def _single_qubit_evolution(pauli, time):
 
 def _two_qubit_evolution(pauli, time, cx_structure):
     # get the Paulis and the qubits they act on
-    labels_as_array = np.array(list(pauli.to_label()))
+    labels_as_array = np.array(list(reversed(pauli.to_label())))
```

Comment:
This is largely irrelevant to the PR, but is weird stuff going to happen if the Pauli has a phase?  With the reversal, it should all get pushed to the end and not cause issues, but in the previous form it feels like it could have done super weird stuff.  Perhaps a test of that would be good if it's intended to work, and if it's meant to be forbidden, then perhaps some error handling in the caller (also with a test)?

Reply:
Indeed weird things can happen, good catch!

Paulis with a phase are okay since all operators are converted to a `SparsePauliOp` and the phase goes into the coefficients of that object. However we currently don't check for complex coefficients and the imaginary parts just get discarded. 

For example, something like `Pauli("iZ")` get's converted to `SparsePauliOp(["Z"], coeffs=[1j])` and that coefficient is converted to 0. So we could definitely add a check for complex coefficients and raise there.

Reply:
I added a check in 4d76180eb.